mongodb/src/test/java/org/hibernate/ogm/datastore/mongodb/test/id/objectid/Bar.java

```diff
@@ -72,4 +73,31 @@ public void setMusicGenre(MusicGenre musicGenre) {
 	public void setDoorMen(Set<DoorMan> doorMen) {
 		this.doorMen = doorMen;
 	}
+
+	@Override
+	public int hashCode() {
+		final int prime = 31;
+		int result = 1;
+		result = prime * result + ( ( doorMen == null ) ? 0 : doorMen.hashCode() );
+		result = prime * result + ( ( musicGenre == null ) ? 0 : musicGenre.hashCode() );
+		result = prime * result + ( ( name == null ) ? 0 : name.hashCode() );
```

Comment:
Note that in the tests, you can also use `Objects#hash(Object...)`. Not that I require you to change it here, just for next time.

It's not always recommended in our internal code as it allocates an array but for the tests, I think it's good enough.